samples/LiveEffect/src/main/cpp/LiveEffectEngine.cpp

```diff
@@ -43,19 +41,25 @@ bool LiveEffectEngine::isAAudioSupported() {
     oboe::AudioStreamBuilder builder;
     return builder.isAAudioSupported();
 }
+
 bool LiveEffectEngine::setAudioApi(oboe::AudioApi api) {
     if (mIsEffectOn) return false;
 
     mAudioApi = api;
     return true;
 }
-void LiveEffectEngine::setEffectOn(bool isOn) {
+
+bool LiveEffectEngine::setEffectOn(bool isOn) {
+    bool success = true;
     if (isOn != mIsEffectOn) {
-        mIsEffectOn = isOn;
         if (isOn) {
-            openStreams();
-            mFullDuplexPass.start();
+            success = openStreams() == oboe::Result::OK;
+            if (success) {
+                mFullDuplexPass.start();
+                mIsEffectOn = isOn;
+            }
         } else {
+            mIsEffectOn = isOn;
```

Comment:
nit:  if possible, move this one down to the last one: after closeStream(mRecordingStream);? ( only change the state after   everything is done )